src/hotspot/share/opto/vectornode.hpp

```diff
@@ -1819,4 +1819,49 @@ class ExpandBitsVNode : public VectorNode {
   virtual int Opcode() const;
 };
 
+//------------------------------LoopVectorMaskNode------------------------------
+// Node for generating a loop vector mask from an integer range
+class LoopVectorMaskNode : public TypeNode {
+ private:
+  int _max_trips;
```

Comment:
Add comment: what is this for exactly? Maybe consider adding more elaborate specification/description above the 3 node classes.

General code style: I think we are trying to get away from the `//--------------NodeName/FunctionName-------` tags, so no need to add them anymore.

Reply:
That is already much better. Could you please also explain what the inputs mean and do?

Reply:
Ok, will do that later